src/module/system/action-macros/athletics/combat-maneuver.ts

```diff
@@ -0,0 +1,67 @@
+import type { ActorPF2e } from "@actor";
+import { SingleCheckActionUseOptions, SingleCheckActionVariant } from "@actor/actions/index.ts";
+import { ItemPF2e, WeaponPF2e } from "@item";
+import { WeaponTrait } from "@item/weapon/index.ts";
+import { ActionMacroHelpers } from "@system/action-macros/index.ts";
+import { CheckResultCallback } from "@system/action-macros/types.ts";
+
+interface CombatManeuverActionUseOptions {
+    /**
+     * The UUID or item instance of the weapon used to perform the disarm maneuver with, if any. Explicitly setting it
+     * to false will disable the automatic attempted lookup of a wielded weapon suitable to use for the action.
+     */
+    item: EmbeddedItemUUID | WeaponPF2e<ActorPF2e> | false;
+}
+
+function isOwnedItemUUID(uuid: EmbeddedItemUUID | string, actor: ActorPF2e) {
+    return uuid.match(/^Actor\.[a-zA-Z0-9]+\.Item\.[a-zA-Z0-9]+$/) && uuid.startsWith(`Actor.${actor.id}`);
+}
+
+abstract class CombatManeuverActionVariant extends SingleCheckActionVariant {
+    static resolveWieldedWeapon(
+        actor: ActorPF2e,
+        original: EmbeddedItemUUID | WeaponPF2e<ActorPF2e> | false | undefined,
+        traits: WeaponTrait | WeaponTrait[],
+    ): WeaponPF2e<ActorPF2e> | undefined {
+        if (original === false) {
+            return undefined;
+        }
+        traits = Array.isArray(traits) ? traits : [traits];
+        let weapon: WeaponPF2e<ActorPF2e> | undefined;
+        if (original && typeof original === "string") {
+            if (isOwnedItemUUID(original, actor)) {
+                const item = fromUuidSync<ItemPF2e<ActorPF2e>>(original);
+                if (item && item.isOfType("weapon")) {
+                    weapon = item;
+                }
+            }
+        } else {
+            weapon = original;
+        }
+        if (weapon && weapon.isOfType("weapon") && weapon.parent?.id === actor.id) {
+            if (!weapon.traits.intersects(new Set<WeaponTrait>(traits))) {
+                console.warn(
+                    `Specified weapon '${weapon.name}' lacks any of the required traits: ${traits
+                        .map((trait) => game.i18n.localize(CONFIG.PF2E.weaponTraits[trait]))
+                        .sort()
+                        .join(", ")}`,
+                );
+            }
+        }
+        if (!weapon) {
+            weapon = traits
+                .flatMap((trait) => ActionMacroHelpers.getApplicableEquippedWeapons(actor, trait) ?? [])
+                .shift();
+        }
+        return weapon;
+    }
+
+    override async use(
+        options: Partial<SingleCheckActionUseOptions & CombatManeuverActionUseOptions> = {},
+    ): Promise<CheckResultCallback[]> {
```

Comment:
Why do you need to subclass use? If its because of the params, couldn't it be part of the function generics?

Reply:
For compile-time type safety basically. It's redefined one more time in each of the specific actions for the same reason.